.travis.yml

```diff
@@ -7,31 +7,50 @@ branches:
     - development
     - testing
 
-addons:
-  apt:
-    update: true
-    packages:
-      - bash
-      - coreutils
-      - gawk
-      - grep
-      - iotop
-      - iproute2
-      - elinks
-      - procps
-      - psmisc
-      - sysstat
+matrix:
+  include:
+    - name: "CentOS 6"
+      env:
+        - DISTRO='centos:6'
+    - name: "CentOS 7"
+      env:
+        - DISTRO='centos:7'
+    - name: "Ubuntu 14.04 (trusty)"
+      env:
+        - DISTRO='ubuntu:14.04'
+    - name: "Ubuntu 16.04 (xenial)"
+      env:
+        - DISTRO='ubuntu:16.04'
+    - name: "Ubuntu 18.04 (bionic)"
+      env:
+        - DISTRO='ubuntu:18.04'
+    - name: "Debian 8 (jessie)"
+      env:
+        - DISTRO='debian:8'
+    - name: "Debian 9 (stretch)"
+      env:
+        - DISTRO='debian:9'
+    - name: "Debian (buster)"
+      env:
+        - DISTRO='debian:buster'
 
-script:
-  - sudo make PREFIX="/usr" install
-  - sudo sed -i 's/^#\(USEPLUGINS\)=.*$/\1="yes"/' /etc/recap.conf
+before_install:
+  - docker pull ${DISTRO}
+
+install:
   - |
-    for plugin in $(ls /usr/lib/recap/plugins-available/); do
-      sudo ln -s \
-        /usr/lib/recap/plugins-available/${plugin} \
-        /usr/lib/recap/plugins-enabled/${plugin}
-    done
-  - sudo recap
-  - sudo cat /var/log/recap/recap.log
+    docker run \
+      --detach \
+      --rm \
+      --tty \
+      --name recap_on_${DISTRO/:/_} \
+      --mount type=bind,src="$(pwd)",dst=/recap \
+      ${DISTRO}
+  - docker exec recap_on_${DISTRO/:/_} /recap/tests/install_deps.sh ${DISTRO}
+
+script:
+  - docker exec recap_on_${DISTRO/:/_} /recap/tests/test_install_recap.sh
+  - docker exec recap_on_${DISTRO/:/_} recap
```

Comment:
In the build this succeeds, but for testing it should [fail here]( https://travis-ci.org/rackerlabs/recap/jobs/456137856#L629-L630 ) if any error occurs.

Reply:
Maybe, what I'm thinking is 
1. Figure out the reason why iotop fails (I think this is related to the environment, container)
2. Add tests to validate each report(?), plugin(?)